service/matching/tasklist/task_completer.go

```diff
@@ -189,16 +189,25 @@ func isDecisionTaskStarted(task *InternalTask, workflowExecutionResponse *types.
 		return true
 	}
 
+	logger.Debug(fmt.Sprintf("Decision task not started. TaskID: %v, ScheduleID: %v, PendingDecisionScheduleID: %v, PendingDecisionState: %v",
+		task.Event.TaskID, task.Event.ScheduleID, workflowExecutionResponse.PendingDecision.ScheduleID, *workflowExecutionResponse.PendingDecision.State),
+		tag.WorkflowID(task.Event.WorkflowID), tag.WorkflowRunID(task.Event.RunID))
+
 	return false
 }
 
-func isActivityTaskStarted(task *InternalTask, workflowExecutionResponse *types.DescribeWorkflowExecutionResponse) bool {
+func isActivityTaskStarted(task *InternalTask, workflowExecutionResponse *types.DescribeWorkflowExecutionResponse, logger log.Logger) bool {
 	// if the scheduleID is different from all pending tasks' scheduleID or the pending activity has PendingActivityStateStarted, then the activity task with the task's scheduleID has already been started
 	for _, pendingActivity := range workflowExecutionResponse.PendingActivities {
 		if task.Event.ScheduleID == pendingActivity.ScheduleID {
 			if *pendingActivity.State == types.PendingActivityStateStarted {
 				return true
 			}
+
+			logger.Debug(fmt.Sprintf("Activity task not started. TaskID: %v, ScheduleID: %v, PendingActivityState: %v",
```

Comment:
prefer to not interpolate logs since they're hard to group. If you can add those values as tags?